content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+
+
+Introduction
+============
+
+The past two weeks we have been traveling and meeting a bunch of
+developers.
+
+It was a chance for us to tell them about Elm through talks at
+conferences and to present a short one-hour workshop.
+
+You can find the slides of the two talks here:
+
+- `a short introduction to Elm by Niko <http://slides.com/n1k0/elm>`_
+- `Talking JSON with Elm by Natim <http://natim.github.io/django-slides/elm-json/>`_
+
+However, this article is a summary of the take aways of this workshop
+for those who would like to know a bit more about Elm.
+
+
+What is Elm and why should I be interested?
+===========================================
+
+Elm is a toolkit that is focused on making sure your app state and the
+HTML that reflects it are always in sync.
+
+If you have been doing any frontend development during the last couple
+of years, you might have seen that the JavaScript ecosystem is a bit
+wild. Keeping up-to-date with the growing number of competing
+frameworks can take a lot of energy, not to mention trying your app on
+all the available browser versions to make sure it works well
+everywhere. Training a team to build an app with React.js,
+Angular2.js, Vue.js or Ember.js as well as looking for the right
+boilerplate to get started with your project are good examples of how
+hight the frustration that the constant reinvention of the ecosystem
+causes can be.
+
+On the other hand, writing tests for the front-end can be tedious and
+supporting Flow or TypeScript can prove to be yet another layer to
+handle in your stack.
+
+If you are already up-to-speed with all those stacks, then you should
+probably keep using them. However if you don't know which one to
+choose and you are looking for a painless way to build frontend
+applications, read on.
+
+
+How is Elm different?
+=====================
+
+First of all, using Elm, the two lines it takes to load your app is
+all the JavaScript you will write. Your application will be written in
+Elm and compiled into JavaScript.
+
+All the mistakes you will do will be caught at compilation, which
+means that you will not have errors when running your app in the
+browser.
+
+Because you will have a compilation step, Elm will force you to handle
+every possible case of your model state. It will make sure you covered
+them on the rendering side.
+
+The tooling is great. ``elm-live`` will automatically update the
+browser code to reflect your code update in realtime. You have a
+debugger that will show you the list of events and show you the state
+of your application at a given time. You can even go back in time and
+replay events. ``elm-format`` will also format your code automatically
+to avoid fighting over coding styles.
+
+
+Getting started
+===============
+
+Installing Elm
+--------------
+
+First of all install ``elm`` and ``elm-format`` using npm the node
+package manager:
+
+.. code-block:: bash
+
+    npm install -g elm elm-format
+
+What is npm? How to install it?
```

Comment:
Is this really necessary? Just linking `npm` to https://www.npmjs.com/ would be just good enough in my opinion.